crates/context_aware_config/src/api/context/handlers.rs

```diff
@@ -921,3 +935,84 @@ async fn priority_recompute(
     ));
     Ok(http_resp.json(response))
 }
+
+#[put("/weight/recompute")]
+async fn weight_recompute(
+    state: Data<AppState>,
+    custom_headers: CustomHeaders,
+    db_conn: DbConnection,
+    #[cfg(feature = "high-performance-mode")] tenant: Tenant,
+    _user: User,
+) -> superposition::Result<HttpResponse> {
+    use superposition_types::cac::schema::contexts::dsl::*;
+    let DbConnection(mut conn) = db_conn;
+
+    let result: Vec<Context> = contexts.load(&mut conn).map_err(|err| {
+        log::error!("failed to fetch contexts with error: {}", err);
+        unexpected_error!("Something went wrong")
+    })?;
+
+    let dimension_data = get_dimension_data(&mut conn)?;
+    let dimension_data_map = get_dimension_data_map(&dimension_data)?;
+    let mut response: Vec<WeightRecomputeResponse> = vec![];
+    let tags = parse_config_tags(custom_headers.config_tags)?;
+
+    let update_contexts = result
+        .clone()
+        .into_iter()
+        .map(|context| {
+            let new_weight = calculate_context_weight(
+                &Value::Object(context.value.clone().into()),
+                &dimension_data_map,
+            )
+            .map_err(|err| {
+                log::error!("failed to calculate context priority: {}", err);
+                unexpected_error!("Something went wrong")
+            });
+
+            match new_weight {
+                Ok(val) => {
+                    response.push(WeightRecomputeResponse {
+                        id: context.id.clone(),
+                        condition: context.value.clone(),
+                        old_weight: context.weight.clone(),
+                        new_weight: val.clone(),
+                    });
+                    Ok(Context {
+                        weight: val,
+                        ..context.clone()
+                    })
+                }
+                Err(e) => Err(e),
+            }
+        })
+        .collect::<superposition::Result<Vec<Context>>>()?;
+
+    let config_version_id =
+        conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
+            let insert = diesel::insert_into(contexts)
+                .values(&update_contexts)
+                .on_conflict(id)
+                .do_update()
+                .set(weight.eq(excluded(weight)))
```

Comment:
here, we also need to update `last_modified_at` and `last_modified_by` columns also